### PR TITLE
feat: add ErrorDetector, added unit-tests for containsIllegalCharacte…

### PR DESCRIPTION
**Pull request by G3nD4**
…rs and startWithDigit methods

### CODE REVIEW

File: src/Tests/ErrorDetectorTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.junit.jupiter.api.Test;
+
+import main.ErrorDetector;
+import main.LexicalErrorType;
+
+import java.lang.reflect.Method;
+import static org.junit.jupiter.api.Assertions.*;
+
+// Your test class
+public class ErrorDetectorTest {
```
Comment (G3nD4) [inline src/Tests/ErrorDetectorTest.java:12]: Need to add tests for this class

**Comment by G3nD4** [comment]
Need to update tests: add cases with lexical errors and assertions for them

File: src/Error_detector/ErrorDetector.java
Comment (Kamil116) [inline src/Error_detector/ErrorDetector.java:1]: Need to check token type. For example, for DATA_TYPE_INTEGER it is ok to start with integer

**Review by Kamil116 — APPROVED** [review]
Need small changes, I will do it

**Pull request merged**